py/desiutil/dust.py

```diff
@@ -189,7 +192,17 @@ def __init__(self, mapdir=None, north="SFD_dust_4096_ngp.fits",
                  south="SFD_dust_4096_sgp.fits", scaling=1.):
 
         if mapdir is None:
-            mapdir = os.environ.get('DUST_DIR', '')
+            dustdir = os.environ.get('DUST_DIR')
+            if dustdir is None:
+                log.critical('Pass mapdir or set $DUST_DIR')
+                raise IOError
```

Comment:
Typically when raising an exception, you pass a message to it.  In this case, the same message as the log message will do.

Also `IOError` might not be ideal here because the exception hierarchy is different between 2 & 3.  I recommend carefully reading https://docs.python.org/3/library/exceptions.html to see if that's what you really want.  For example, `ValueError` might be just as appropriate.

Please also make sure that the unit tests actually test the case where an invalid value is passed during initialization.

Reply:
I now raise a ValueError with a message, and I've added a unit test to check that ValueErrors are thrown appropriately when `DUST_DIR` doesn't exist or when a meaningless map directory is passed.